Split the `connections_are_properly_closed` test into two

hrxi opened the pull request:
The problem with the old one was that the peers re-discovered each other
after being disconnected, hence the
`assert_eq!(net.get_peers().len(), 0)` failed sometimes.

Fix this by splitting it up into two tests, one where we drop the
assertions and one where we disconnect the networks so that the peers
can't find each other again.

Fixes #708.

jsdanielh commented on network-libp2p/tests/network.rs:
```diff
@@ -405,15 +405,33 @@ async fn connections_are_properly_closed() {
+    let net2_peer_id = *net2.local_peer_id();
+    drop(net2);
+
+    peer1.close(CloseReason::Other);
```
I'd first close `peer1` connection and then drop `net2` since `peer1` comes from `net2`.

hrxi replied:
I misidentified whose network `peer1` is, I thought it was a peer of `net1`. I now swapped the roles of `net1` and `net2` and it should be good now?

hrxi committed:
Split the `connections_are_properly_closed` test into two

The problem with the old one was that the peers re-discovered each other
after being disconnected, hence the
`assert_eq!(net.get_peers().len(), 0)` failed sometimes.

Fix this by splitting it up into two tests, one where we drop the
assertions and one where we disconnect the networks so that the peers
can't find each other again.

Fixes #708.
